### PR TITLE
Ask if user want to update home content when available

### PR DESCRIPTION
**Pull request by Atlante45**
This PR adds the feature that checks for a newer home content on sandbox startup and asks the user if he wants to update.

### CODE REVIEW

**Comment by Atlante45** [comment]
Test plan:

- Launch sandbox (content set should get installed).
- Close and restart sandbox (nothing specific should happen).
- Close sandbox.
- Go to "C:/Users/<username>/AppData/Roaming/High Fidelity - PR8656/domain-server/config.json".
- Change the homeContentLastModified value to an old date (ie: 2012 should do it).
- Start sandbox.
- You should get prompted for an update.
- Accept and notice the content set getting re-downloaded.
- Close sandbox.
- Modify the config file again.
- Start sandbox again and say "no" to the first update popup.
- Restart sandbox and notice you do not get prompted again.

**Comment by hifi-gustavo** [comment]
Build console output for commit: cb4d526dcfdbe053330766945fb4f68d6ff1c16a available at: https://gist.github.com/c6663ed4d4cb94b25991e38c23e4620f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8656-cb4d526.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8656-cb4d526.dmg

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**QA Pass**: **QA Pass** FAILURE
**Additional Test Results + Observations**: Fail - ```config.json``` did not have ```homeContentLastModified value```.  I was successfully able to launch sandbox, get new content, close and restart successfully however.

**Comment by cozza13** [comment]
Same issue as stated by @kevinmthomasse

**Comment by Atlante45** [comment]
@kevinmthomasse @cozza13 Dammit, I put the wrong file path in the test plan, my bad.
This is the one: `C:\Users\<username>\AppData\Roaming\High Fidelity\Server Console`

**Comment by kevinmthomasse** [comment]
re-testing

**Comment by kevinmthomasse** [comment]
## QA Re-Test
**QA Tester**: KevinMThomas
**QA Pass**: **QA Pass** FAILURE
**Additional Test Results + Observations**: Re-tested as the - ```config.json``` did not have ```homeContentLastModified value```.  It only had the value ```{"hasRun":true}
I was successfully able to launch sandbox, get new content, close and restart successfully however.  ```json``` file attached as I had to change the file extension to .txt however it is a ```json``` file.
[config.json.txt](https://github.com/highfidelity/hifi/files/491625/config.json.txt)

**Comment by Atlante45** [comment]
@kevinmthomasse I did it again, it's in the PR directory, not the main directory!!! -__-"
`C:\Users\Clement\AppData\Roaming\High Fidelity - PR8656\Server Console\config.json`

**Comment by kevinmthomasse** [comment]
## QA Re-Test
**QA Tester**: KevinMThomas
**QA Pass**: **QA Pass** SUCCESS
**Additional Test Results + Observations**: Re-tested as the - ```config.json``` did have ```homeContentLastModified value``` which was ```{"hasRun":true,"homeContentLastModified":"2016-09-25T11:33:19.663Z"}```

File: server-console/src/main.js [hunk 1]
```diff
@@ -486,6 +486,76 @@ function updateTrayMenu(serverState) {
...
+                      title: 'Are you sure?',
+                      message: 'This action will delete your current sandbox content.\nDo you wish to continue?'
+                  }, function(idx) {
+                    if (idx === 0 && homeServer) {
+                        deleteResourceDirectoriesAndRestart();
+                    }
+                  });
+                } else {
+                  // They don't want to update, mark content set as current
+                  userConfig.set('homeContentLastModified', new Date());
```
Comment (birarda) [inline server-console/src/main.js]: Kind of a bummer that once you say no once you won't see the dialog again until we update the content set on our end - is there anything quick we could do to allow this? Maybe a separate PR that lets you toggle the replacement from a menu item.

File: server-console/src/main.js [hunk 2]
```diff
@@ -486,6 +486,76 @@ function updateTrayMenu(serverState) {
...
+function deleteResourceDirectoriesAndRestart() {
+    homeServer.stop();
+    deleteResourceDirectories();
+    maybeInstallDefaultContentSet(onContentLoaded);
+}
+
+function checkNewContent() {
+    // Start downloading content set
+    var req = request.head({
+        url: "http://cachefly.highfidelity.com/home.tgz"
```
Comment (birarda) [inline server-console/src/main.js]: we should constantize this somewhere else in the same file

**Comment by hifi-gustavo** [comment]
Build console output for commit: f001cf7570150cac91467de64493d885484325c6 available at: https://gist.github.com/c6663ed4d4cb94b25991e38c23e4620f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8656-f001cf7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8656-f001cf7.dmg

**Comment by Atlante45** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: f001cf7570150cac91467de64493d885484325c6 available at: https://gist.github.com/c6663ed4d4cb94b25991e38c23e4620f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8656-f001cf7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8656-f001cf7.dmg

**Pull request merged**